torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -480,7 +549,7 @@ TensorView* arange(Val* start, Val* end, Val* step, DataType dtype) {
                  .contiguity({true})
                  .shape({size})
                  .build();
-  IrBuilder::create<ARangeOp>(out, start, end, step);
+  IrBuilder::create<ARangeOp>(out, start, end, step, dtype);
```

Comment:
All tensor factories need to keep the the dtype, because its `sameAs` need to check the equality of dtype, but `Expr::sameAs` is not checking output